### DISCUSSION TITLE
[Feature] Search for people by facial-reference-photo

### DISCUSSION THREAD

**Discussion by delta0815** [immich-app/immich#9285] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi all,

I hope my idea hasn't been requested in the past. At least I couldn't find it in the board: Based on the existing facial recognition capabilities, I was wondering if there are any plans to extend the search-feature by a "search for people by facial-refernece-photo".

Rational for the request: I currently do a lot of event-photography. Especially for marathons etc. While we do have consent to take pictures of the attendees I would like to simplify the search process for attendees of the events: By uploading a facial reference image, the application should provide all images which match the uploaded picture of the face.

Thanks for your consideration.

Cheers,

Oliver

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
If you upload an image and then open the info pane, it will show the people found in that image.

![image](https://github.com/immich-app/immich/assets/11602424/2802a13f-d88a-4736-9367-4f4379b6b83c)

**Reply by delta0815** [reply]
Oh, maybe my explaination was not specific enough:

I saw that feature and it's amazing, however I was asking for an enhancement of the search capability:

Imagine an event with hundrets images of hunderts of faces: Then a search feature which would allow someone to upload an image of his/her reference face in order to find all images which matches that face. That is actually what I am looking for.

Appologies if my first description wasn't specific enough.

**Reply by NicholasFlamy** [reply]
A way to do that right now would be to upload a new photo of the person (so your facial reference) and go to the info and tap the person detected and see all the photos listed under that person. Obviously it's not a dedicated feature specific to your use case, but it should get the job done and not be too difficult.

**Reply by delta0815** [reply]
That's a great idea of locating the images. I was actually looking for a way to make it a bit more "data-privacy" oriented so that you would only get your personal images if you provide an image from your face. But still: Your solution would at least allow to search for your own images.

Thanks and best regards,

Oliver
